src/wfcomman.c

```diff
@@ -1632,28 +1632,16 @@ AppCommandProc(register DWORD id)
 
    case IDM_FORMAT:
 
-      if (CancelInfo.hCancelDlg) {
-         SetFocus(CancelInfo.hCancelDlg);
-         break;
+      if (!hwndFormatSelect)
+      {
+         hwndFormatSelect = CreateDialog(hAppInstance, (LPTSTR)MAKEINTRESOURCE(FORMATSELECTDLG), 
+            hwndFrame, (DLGPROC)FormatSelectDlgProc);
       }
-
-      if (CancelInfo.hThread) {
-         //
-         // Don't create any new worker threads
-         // Just create old dialog
-         //
-
-         CreateDialog(hAppInstance, (LPTSTR) MAKEINTRESOURCE(CANCELDLG), hwndFrame, (DLGPROC) CancelDlgProc);
-
-         return TRUE;
+      else
+      {
+         SetActiveWindow(hwndFormatSelect);

```

Comment:
This suggested change makes sense, but I'm curious how you are getting into the state where hwndFormatSelect is hidden. I was able to test by forcing the dialog hidden with SW_HIDE, but I'd like to test the scenario without programmatically hiding the window.

Reply:
Actually, thinking through this a bit more... hwndFormatSelect is intentionally hidden while the SHFormatDrive dialog is visible. So, if the user currently has the SHFormatDrive dialog up (say they are formatting a drive) and they then click the "Format" menu item again, do we really want to show hwndFormatSelect? If we do show it, then both hwndFormatSelect and SHFormatDrive  will be visible at the same time, but hwndFormatSelect won't be usable, since it is the parent of the modal SHFormatDrive dialog.